packages/core/src/shared/ui/nestedWizardPrompter.ts

```diff
@@ -0,0 +1,63 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Wizard, WizardOptions } from '../wizards/wizard'
+import { Prompter } from './prompter'
+import { WizardPrompter } from './wizardPrompter'
+import { createHash } from 'crypto'
+
+/**
+ * An abstract class that extends the base Wizard class plus the ability to
+ * use other wizard classes as prompters
+ */
+export abstract class NestedWizard<T> extends Wizard<T> {
```

Comment:
This name seems confusing, unless I'm mistaken. This class is a Wizard that contains nested wizards, it's not itself nested. Right? So maybe a better name is `Wizards` (plural). That also improves discoverability because it sorts alphabetically next to `Wizard`.